### ISSUE TITLE
Things to implement 

### ISSUE THREAD

**Issue by doloresgarcia** [selvaggi/mlpf#1]
- Obtain number of hits and true particles per event distribution (2d histogram) and mean and variance
- check if the energy of the particle matches the energy of the hits that link to the particle and correct it
- train model gravnet, add track p as an input
- implement the plots, per type of particle (as in the paper)
- what hits is it picking as cluster center
- also check if the model would be best if the E pred was calculated from the sum of the particles inside the cluster. 
- implement inference by removing hits that have been already associated as in procedure 2 of the end to end paper

**Comment by doloresgarcia** [comment]
Note: currently the clustering is implemented as x_alpha (so not doing the mean of the particles in the cluster) this could also be done but may be worth it to keep the simple approach for now

**Comment by doloresgarcia** [comment]
**About adding particle ID:** this can not be regressed directly because it should be obtained from the cluster. 
Options:
* to pull the clusters and then regress 1 output for each subgraph
* to do some aggregation from per hit regression (this is more sloppy)

**Comment by doloresgarcia** [comment]
- [ ] higher q value [0.3] EASY 
- [ ] sample different points in the class and add that to the loss Gregor 
- [ ] Take out the batchnom/dropout. EASY  NOW
- [ ] try other networks Dolo 
- [ ] batchnorm before the output layer. NOW
- [ ] plot the gravnet clustering space 
- [ ] Option to start with the physical space [NOT FOR NOW]
